### PR TITLE
feat: heatmap improvements

### PR DESCRIPTION
**Pull request by dogfrogfog**
## Brief
- 

## Changes
- moved heatmap legend
- add loading indicator when fetching profile data after selection
- add basic tooltip with time
- show tooltip when hover on selected area
- add common timestamp format
- fix labels failing request

## Concerns
- same tooltip logic (mouseenter/mouseleave/mousemove events) should be reused between FlamegraphTooltip/TableTooltip and new HeatmapTooltip components

### CODE REVIEW

**Comment by dogfrogfog** [comment]
https://user-images.githubusercontent.com/47758224/190422023-223668ef-7696-4326-a720-0ec07d8cbc06.mov

File: webapp/javascript/components/Heatmap/Heatmap.spec.tsx [hunk 1]
```diff
@@ -50,10 +50,12 @@ describe('Component: Heatmap', () => {
   it('should have correct color scale', () => {
     renderHeatmap();
 
-    const [minTextEl, maxTextEl] = within(
+    const [maxTextEl, midTextEl, minTextEl] = within(
       screen.getByTestId('color-scale')
     ).getAllByRole('textbox');
-    expect(minTextEl.textContent).toBe(heatmapMockData.minDepth.toString());
+    console.log(minTextEl, midTextEl, maxTextEl);
```
Comment (Rperry2174) [inline webapp/javascript/components/Heatmap/Heatmap.spec.tsx:56]: did you mean to keep this here? 
```suggestion
```

**Review by Rperry2174 — APPROVED** [review]
LGTM!

File: webapp/javascript/components/Heatmap/Heatmap.spec.tsx (hunk 1 above)
Comment (dogfrogfog) [reply webapp/javascript/components/Heatmap/Heatmap.spec.tsx:56]: No, that should be removed

**Pull request merged**